File: 251004/Ubozhenko/DistComp_lab_1/DistComp_1/Exceptions/ErrorMessages.cs
namespace DistComp_1.Exceptions;

public static class ErrorMessages
{
    public static string AuthorNotFoundMessage(long id) => $"Author with id {id} was not found.";
    public static string NewsNotFoundMessage(long id) => $"News with id {id} was not found.";
    public static string LabelNotFoundMessage(long id) => $"Label with id {id} was not found.";
    public static string ReactionNotFoundMessage(long id) => $"Reaction with id {id} was not found.";

    public static string AuthorAlreadyExists(string login) => $"Author with login '{login}' already exists.";
    public static string NewsAlreadyExists(string title) => $"News with title '{title}' already exists.";
    public static string LabelAlreadyExists(string tag) => $"Label with name '{tag}' already exists.";
}

File: 251002/Trubach/Discussion/src/common/exceptions/error.messages.ts
export class ErrorMessages {
  static LabelNotFoundMessage(id: number): string {
    return `Label with id ${id} was not found.`;
  }

  static ArticleNotFoundMessage(id: number): string {
    return `Article with id ${id} was not found.`; 
  }

  static NoteNotFoundMessage(id: number): string {
    return `Note with id ${id} was not found.`; 
  }

  static LabelAlreadyExists(name: string): string {
    return `Label with name '${name}' already exists.`;
  }

  static ArticleAlreadyExists(title: string): string {
    return `Article with title '${title}' already exists.`;  
  }

  static NoteAlreadyExists(content: string): string {
    return `Note with content '${content}' already exists.`;  
  }
}
